lib/ain-evm/src/core.rs

```diff
@@ -123,26 +126,46 @@ impl EVMCoreService {
             value,
             data,
             gas_limit,
+            gas_price,
+            max_fee_per_gas,
             access_list,
             block_number,
+            transaction_type,
         } = arguments;
 
-        let (state_root, block_number) = self
+        let (state_root, block_number, beneficiary, base_fee, timestamp) = self
             .storage
             .get_block_by_number(&block_number)?
-            .map(|block| (block.header.state_root, block.header.number))
+            .map(|block| {
+                (
+                    block.header.state_root,
+                    block.header.number,
+                    block.header.beneficiary,
+                    block.header.base_fee,
+                    block.header.timestamp,
+                )
+            })
             .unwrap_or_default();
         debug!(
             "Calling EVM at block number : {:#x}, state_root : {:#x}",
             block_number, state_root
         );
-
-        let vicinity = Vicinity {
+        debug!("[call] caller: {:?}", caller);
+        let vicinity: Vicinity = Vicinity {
```

Comment:
Type can be infer

Reply:
rm and auto infer https://github.com/DeFiCh/ain/pull/2318/commits/e1a169c050ef0cb3ed6a9cb55c99863474209e46